index.bs

```diff
@@ -456,37 +468,30 @@ origin, as described in [[#serializing]].
 
 Issue: TODO(jww): Formal definition of the headers and responses w/grammars.
 Also need to be explicit about `*` having same limitations as
-`Access-Control-Allow-Origin` w/credentials. Also need to describe opt-outs to
-use old-school Origin header and ACAO header.
+`Access-Control-Allow-Origin` w/credentials. 
 
 ## `postMessage` ## {#postmessage-ac}
 
-Cross-origin messaging via <a>`postMessage`</a> provides requires that the
-recipient be able to see the suborigin namespace of the message sender so an
-appropriate access control decision can be made. When a message is sent from a
+Cross-origin messaging via <a>`postMessage`</a> requires that the
+recipient be able to see the suborigin namespace of the message sender so it can make an
+appropriate access control decision. When a message is sent from a
 suborigin namespace, the receiver has the `event.origin` value set to the
 serialized suborigin value instead of the serializied origin, as described in
 [[#serializing]]. Additionally, a new `suborigin` property must be added to the
 `MessageEvent` given to the receiver which contains the suborigin namespace
 value.
 
-Issue: TODO(jww) Add opt-out so Origin header is traditional header.
 
 ## Workers ## {#workers-ac}
 
-Issue: TODO(jww): There are many details and specifics to fill in here.
+User agents MUST refuse to create or execute any workers in a page executing in
+a sub-origin.
 
-Workers [[!WORKERS]] are given the security context of the creating Suborigin
-and all origin checks should check suborigins as well.
-
-The Service Worker [[!SERVICE-WORKERS]] registration API must not be accessible
-from suborigins. As a result of the above restriction on Workers, since a
-Service Worker cannot be created by a Suborigin, no Service Workers will be able
-to intercept the requests of a Suborigin.
 
 Note: This may change in the future, and Suborigins may eventually be allowed to
-register Service Workers, but for now, allowing the creation of Service Workers
-from Suborigins adds too many complications.
+register Service Workers, but for now, allowing the creation of (Service)
```

Comment:
I find "(Service) Workers" awkward. Maybe "all Workers, including Service Workers and Shared Workers," instead?